clusterloader2/pkg/measurement/common/network-policy/network-policy-enforcement-latency.go

```diff
@@ -0,0 +1,568 @@
+	klog.V(2).Infof("Creating test client deployments for measurement %q", networkPolicyEnforcementName)
+	templateMap["TypeLabelValue"] = testType
+
+	// Create a test client deployment for each test namespace.
+	for i, ns := range nps.targetNamespaces {
+		templateMap["Name"] = fmt.Sprintf("%s-%s-%d", testType, netPolicyTestClientName, i)
+		templateMap["TargetNamespace"] = ns
+		templateMap["AllowPolicyName"] = fmt.Sprintf("%s-%d", allowPolicyName, i)
+
+		if err := nps.framework.ApplyTemplatedManifests(manifestsFS, deploymentFilePath, templateMap); err != nil {
+			return fmt.Errorf("error while creating test client deployment: %v", err)
+		}
+	}
+
+	return nil
+}
+
+func (nps *networkPolicyEnforcementMeasurement) createLoadPolicies(config *measurement.Config) {
+	policyLoadTargetBaseName, err := util.GetStringOrDefault(config.Params, "policyLoadTargetBaseName", defaultPolicyTargetLoadBaseName)
+	if err != nil {
+		klog.Errorf("Failed getting parameter policyLoadBaseName value, error: %v", err)
+		return
+	}
+
+	policyLoadCount, err := util.GetIntOrDefault(config.Params, "policyLoadCount", defaultPolicyLoadCount)
+	if err != nil {
+		klog.Errorf("Failed getting parameter policyLoadBaseName value, error: %v", err)
+		return
+	}
+
+	policyLoadQPS, err := util.GetIntOrDefault(config.Params, "policyLoadQPS", defaultPolicyLoadQPS)
+	if err != nil {
+		klog.Errorf("Failed getting parameter policyLoadQPS value, error: %v", err)
+		return
+	}
+
+	policiesPerNs := policyLoadCount / len(nps.targetNamespaces)
+	qpsSleepDuration := (1 * time.Second) / time.Duration(policyLoadQPS)
+
+	for nsIdx, ns := range nps.targetNamespaces {
+		baseCidr := fmt.Sprintf("10.0.%d.0/24", nsIdx)
+
+		for depIdx := 0; depIdx < policiesPerNs; depIdx++ {
+			// This will be the same as "small-deployment-0".."small-deployment-50",
+			// that is used in the ClusterLoader2 load test.
+			// https://github.com/kubernetes/perf-tests/tree/master/clusterloader2/testing/load
+			podSelectorLabelValue := fmt.Sprintf("policy-load-%s-%d", policyLoadTargetBaseName, depIdx)
+			templateMapForTargetPods := map[string]interface{}{
+				"Name":                  fmt.Sprintf("%s-%d", podSelectorLabelValue, nsIdx),
+				"Namespace":             ns,
+				"PodSelectorLabelKey":   "name",
+				"PodSelectorLabelValue": podSelectorLabelValue,
+				"CIDR":                  baseCidr,
+			}
+
+			go func() {
+				if err := nps.framework.ApplyTemplatedManifests(manifestsFS, policyLoadFilePath, templateMapForTargetPods); err != nil {
+					klog.Errorf("Error while creating load network policy for label selector 'name=%s': %v", podSelectorLabelValue, err)
+				}
+			}()
+
+			time.Sleep(qpsSleepDuration)
+		}
+	}
+
+	expectedFinishTime := time.Now().Add(time.Duration(policyLoadCount/policyLoadQPS) * time.Second)
+	// Add a short time buffer before completing, to not affect next test steps.
```

Comment:
Hm, maybe let's switch to rate.NewLimiter as mentioned above and make api calls inline instead of separate goroutine and let's get rid of this part.

Reply:
Thanks. Done.